ssl/ssl_lib.c

```diff
@@ -3600,6 +3600,13 @@ const SSL_CIPHER *SSL_get_current_cipher(const SSL *s)
     return (NULL);
 }
 
+const SSL_CIPHER *SSL_get_pending_cipher(const SSL *s)
+{
+    if (s->s3->tmp.new_cipher != NULL)
+        return s->s3->tmp.new_cipher;
+    return NULL;
```

Comment:
Good point.  I think when I started writing this I figured I would need to check one of the things being dereferenced as well, but all SSL_METHOD ssl_new functions are or call ssl3_new, which allocates s->s3.